src/Server/HTTPHandler.cpp

```diff
@@ -968,6 +981,14 @@ HTTPRequestHandlerFactoryPtr createPredefinedHandlerFactory(IServer & server,
     Poco::Util::AbstractConfiguration::Keys headers_name;
     config.keys(config_prefix + ".headers", headers_name);
 
+    std::optional<BasicCredentials> config_credentials;
+    if (config.has(config_prefix + ".handler.user") || config.has(config_prefix + ".handler.password"))
+    {
+        config_credentials.emplace(
+            config.getString(config_prefix + ".handler.user", "default"),
+            config.getString(config_prefix + ".handler.password", ""));
+    }
```

Comment:
Maybe it's better to invent a new structure named e.g. `HTTPHandlerConfig`:

```
struct HTTPHandlerConfig
{
    std::optional<BasicCredentials> credentials;

    /// TODO:
    /// String quota;
    /// String default_database;

    HTTPHandlerAuthConfig(const Poco::Util::AbstractConfiguration & config, const std::string & config_prefix)
};
```
and use it in both `createDynamicHandlerFactory` and `createPredefinedHandlerFactory`?

Reply:
Done

_Though structures that accept `config` and parsing it inside looks odd to me, since they makes the config does not looks like a config at all, even though this is already used in tons of places_